app/api/getTokenPrice/route.ts

```diff
@@ -0,0 +1,49 @@
+import { NextRequest, NextResponse } from 'next/server';
+
+/**
+ * @description get the price of the token
+ * 
+ * @param {coins, fiats}
+ * @sets price of the token in dollars
+ */
+async function getTokenPrices({ coins, fiats }: { coins: string; fiats: string }) {
+  try {
+    const response = await fetch(
+      `https://pro-api.coingecko.com/api/v3/simple/price?ids=${coins}&vs_currencies=${fiats}`,
+      {
+        headers: {
+          'X-CG-Pro-API-Key': process.env.COINGECKO_API_KEY || '',
+        }
+      }
+    );
+    if (response.status !== 200) {
+      throw new Error(`Failed to fetch token price ${response.status} ${response.statusText}`);
+    }
+    const data = await response.json();
+    return data;
+  } catch (error) {
+    console.error('Error fetching token price:', error);
+    throw error;
+  }
+}
+
+export async function GET(req: NextRequest) {
```

Comment:
How is the route getting set to api/getTokenPrices ?

Reply:
that's handled by next.js apis